nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java

```diff
@@ -185,7 +185,7 @@ public void process(final InputStream in) throws IOException {
         // if TCP and we don't end in a new line then add one
         final String protocol = context.getProperty(PROTOCOL).getValue();
         if (protocol.equals(TCP_VALUE.getValue())) {
-            final byte[] buf = baos.getUnderlyingBuffer();
+            final byte[] buf = baos.toByteArray();
```

Comment:
I think we need to get the buffer after we potentially change the newline.  If we do this here then change the newline we won't see it since now we're getting a copy of the buffer instead of the actual underlying buffer.